packages/admin/dashboard/src/routes/customers/customer-detail/components/customer-address-section/customer-address-section.tsx

```diff
@@ -0,0 +1,92 @@
+import { HttpTypes } from "@medusajs/types"
+import { Container, Heading, toast, usePrompt } from "@medusajs/ui"
+import { useTranslation } from "react-i18next"
+
+import { Trash } from "@medusajs/icons"
+import { Link, useNavigate } from "react-router-dom"
+import { ActionMenu } from "../../../../../components/common/action-menu"
+import { Listicle } from "../../../../../components/common/listicle"
+import { useDeleteCustomerAddress } from "../../../../../hooks/api/customers"
+
+type CustomerAddressSectionProps = {
+  customer: HttpTypes.AdminCustomer
+}
+
+export const CustomerAddressSection = ({
+  customer,
+}: CustomerAddressSectionProps) => {
+  const { t } = useTranslation()
+  const prompt = usePrompt()
+  const navigate = useNavigate()
+  const { mutateAsync: deleteAddress } = useDeleteCustomerAddress(customer.id)
+
+  const addresses = customer.addresses
+
+  const handleDelete = async (address: HttpTypes.AdminCustomerAddress) => {
+    const confirm = await prompt({
+      title: t("general.areYouSure"),
+      description: t("general.areYouSureDescription", {
+        entity: t("fields.address"),
+        title: address.address_name ?? "n/a",
+      }),
+      verificationInstruction: t("general.typeToConfirm"),
+      verificationText: address.address_name ?? "address",
+      confirmText: t("actions.delete"),
+      cancelText: t("actions.cancel"),
+    })
+
+    if (!confirm) {
+      return
+    }
+
+    await deleteAddress(address.id, {
+      onSuccess: () => {
+        toast.success(
+          t("general.success", { name: address.address_name ?? "address" })
+        )
+
+        navigate(`/customers/${customer.id}`, { replace: true })
+      },
+      onError: (e) => {
+        toast.error(e.message)
+      },
+    })
+  }
+
+  return (
+    <Container className="p-0">
+      <div className="flex items-center justify-between px-6 py-4">
+        <Heading level="h2">{t("addresses.title")}</Heading>
+        <Link to={`create-address`} className="text-ui-fg-muted text-xs">
+          Add
+        </Link>
+      </div>
+
```

Comment:
nit/suggestion: add placeholder UI for empty state

Reply:
@fPolic added